### ISSUE TITLE
Incorrect ILMerge in NuGet 2.8.6 introduces breaking dependency to .NET 4.5

### ISSUE THREAD

**Issue by PhilipAtz** [NuGet/Home#1611]
When NuGet 2.8.6 was released, even though it was a patch release, it changed the target framework from .NET 4.0 to 4.5. However, due to an incorrect ILMerge option, types are not translated correctly between System.Core.dll and mscorlib.dll. This causes machines (like old build servers) that don't have .NET 4.5 installed to not be able to run NuGet 2.8.6.

Error message:
Could not load type 'System.Runtime.CompilerServices.ExtensionAttribute' from assembly 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.

See relevant posts on StackOverflow:
http://stackoverflow.com/questions/13748055/could-not-load-type-system-runtime-compilerservices-extensionattribute-from-as

http://stackoverflow.com/questions/31681248/nuget-2-8-6-causing-dependency-on-net-4-5

**Comment by emgarten** [comment]
Thank you for the information. The workaround for this is to use the NuGet.CommandLine 2.8.5 package which can be found on nuget.org. We are currently are working on nuget.exe 3.3 and are not planning on releasing that for .net 4.0

**Comment by Myster** [comment]
There is a comment on an SO answer that says this is fixed in nuget 2.9.
Is that true?
If true, should this be marked as fixed in that version rather than won't fix? 
(SO answer: http://stackoverflow.com/a/33226602/74449 )

**Comment by Myster** [comment]
I don't think the suggested work-around will work. 
On a build agent without .net 4.5 installed I have this error:

```
The 'Microsoft.Bcl.Build 1.0.14' package requires NuGet client version '2.8.1' or above, but the current NuGet version is '2.7.40906.75'.
```

When I tell TeamCity to use 2.8.1 I get this error:

```
[12:38:13][install] Starting NuGet.exe 2.8.50320.36 from D:\BuildAgent\tools\NuGet.CommandLine.2.8.1.nupkg\tools\NuGet.exe
[12:38:13][install] Could not load type 'System.Runtime.CompilerServices.ExtensionAttribute' from assembly 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.
```
